### DISCUSSION TITLE
[组件:下载弹幕](已修复)可能是一个导致彩色弹幕变成白色的BUG？

### DISCUSSION THREAD

**Discussion by Tinhone** [the1812/Bilibili-Evolved#3960] [综合讨论 / General]
组件: 下载弹幕
Commit: 3397e56dc
内部名称: downloadDanmaku

下载的ASS格式弹幕貌似因为格式错误，而导致彩色弹幕变成白色
通过组件下载获取的无法显示的版本：`{\pos(998.5,450)\C&H1200E7&}猝死的孩子有番看`
可以被显示的版本：`{\pos(998.5,450)\c&H1200E7&}猝死的孩子有番看`

使用的播放器是没有加插件的 Potplayer

因为不清楚是否为个人播放器问题，因此来发个讨论询问一下各位大佬们，上面提到的情况是否能在其他播放器复现

https://github.com/the1812/Bilibili-Evolved/blob/master/registry/lib/components/video/danmaku/converter/ass-danmaku.ts#L50`
https://github.com/the1812/Bilibili-Evolved/blob/master/registry/lib/components/video/danmaku/converter/danmaku-converter.ts#L104
https://github.com/the1812/Bilibili-Evolved/blob/master/src/components/video/ass-utils.ts#L16
顺着找下来发现，小写c被转换成大写C的原因是`ass-utils.ts#L16`的`.toUpperCase()`

**Comment by the1812** [comment]
应该是我写迷糊了 (当时只想转换一下 HEX 颜色代码)

**Reply by Tinhone** [reply]
这样啊，那还需要再开一个issue吗

**Reply by the1812** [reply]
不用了
